calyx-opt/src/passes/comb_prop.rs

```diff
@@ -288,6 +290,58 @@ impl CombProp {
         }
     }
 
+    fn parent_is_one_bit_one(parent: &ir::PortParent) -> bool {
```

Comment:
Use `Guard::is_true` instead. If it doesn't implement the correct check, please generalize it to do so. We can implement a similar method on ports.